xapian-core/queryparser/termgenerator_internal.cc

```diff
@@ -229,9 +270,23 @@ parse_terms(Utf8Iterator itor, unsigned cjk_flags, bool with_positions,
 	    do {
 		Unicode::append_utf8(term, ch);
 		prevch = ch;
-		if (++itor == Utf8Iterator() ||
-		    (cjk_flags && CJK::codepoint_is_cjk(*itor)))
+		if (++itor == Utf8Iterator())
 		    goto endofterm;
+		else if (cjk_flags) {
+		    // Only deal mixed Chinese numbers which start
```

Comment:
Stylistically, if you have braces `{`...`}` around one branch of an `if`, it's preferable to add them to all the branches.

However, here the `if` part transfers the flow of control away, so there's no need for the `else` anyway (and the Xapian style guide suggests omitting it in such cases).